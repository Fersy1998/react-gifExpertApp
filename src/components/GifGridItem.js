import React from 'react'

import PropTypes from 'prop-types'
const GifGridItem = ({title, url}) => {
    //console.log(id +url+title);
  return (
    <div className="card animate__animated animate__fadeInLeft" >
        <img alt={title} src={url} /> 
        <p>{title}</p>
    </div>
  )
  
}


GifGridItem.propTypes = {
    title:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired
}
export default GifGridItem;


