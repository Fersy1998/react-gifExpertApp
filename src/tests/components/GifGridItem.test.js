import React from 'react';

//import { shallowToJson } from "enzyme-to-json";
import {shallow} from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Pruebas en GifGridItem", ()=>{

    const wrapper=shallow(<GifGridItem title='Hola' url="https://r34663.jpg"/>);
    test('should show the component', () => {
        
        expect(wrapper).toMatchSnapshot();
        
        
    })
    test('should say the </p> is ok', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe('Hola');
    })
    
    test('should have the img and url in <img>', () => { 
        const img=wrapper.find('img');
        expect(img.props().src).toBe('https://r34663.jpg');
        expect(img.props().alt).toBe('Hola');
    })
    test('should have the class animate__fadeInLeft', () => { 
        const div=wrapper.find('div');
        expect(div.props().className).toContain("animate__fadeInLeft");
        
        //El renglón de arriba se puede hacer también
        
        /*
        const className=div.prop('className');
        expect(className.includes("animate__fadeInLeft")).toBe(true);
        
        
        */
    
    })
})