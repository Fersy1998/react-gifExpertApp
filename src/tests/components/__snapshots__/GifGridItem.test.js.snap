// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem should show the component 1`] = `
<div
  className="card animate__animated animate__fadeInLeft"
>
  <img
    alt="Hola"
    src="https://r34663.jpg"
  />
  <p>
    Hola
  </p>
</div>
`;
